### PR TITLE
[TX Builder] Transactions are initialised when entering a contract with a wrong address (Abi not retrieved properly) 

### LINKED ISSUE #141: Transaction builder forgets added transactions when adding new ABI or address
Hi there, 

This is the issue is occurring on:

app url:
https://gnosis-safe.io/app/#/safes/0x0013CEFc104698Bb40d574E2aE6C822d5D52cff3/apps?appUrl=https://cloudflare-ipfs.com/ipfs/QmTJMCBZHX56z36aTUaL2QZ3rKTkzaq1CG3zhPSKb1Su3a

Brave version:
Release Notes V1.26.74 (June 30, 2021)

Description:

After the first transaction has been added, the front end reflects this by showing (1) in the send transactions button. Then if a new ABI or Address is interested in to the "Enter Address or ABI" field, the UI reverts to "Send transactions" without the 1. 

Please find the video attached. 

https://user-images.githubusercontent.com/7915365/124475485-0f88dc00-dda2-11eb-8aa0-1cd1a6cc7e73.mov

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
Resolves #141 

## How this PR fixes it
The problem was the component Builder being destroyed when the contract address entered is wrong. We moved the transaction management to the parent so they are kept

## How to test it
1) Open the transaction builder inside a Safe web and add a correct contract address for example 0x39aa39c021dfbae8fac545936693ac917d5e7563 (`mainnet`)
2) Add transactions using the `redeemUnderlying` method for example like in the video in #141 
3) Now enter a wrong address (For example deleting some characters) and then a correct one again
4) Transactions should be kept and when trying to submit the data associated should be ok

### CODE REVIEW

File: apps/tx-builder/src/components/Dashboard.tsx [hunk 1]
```diff
@@ -69,6 +70,35 @@ const Dashboard = () => {
...
+    [transactions],
+  );
+
+  const handleSubmitTransactions = useCallback(async () => {
+    if (!transactions.length) {
+      return;
+    }
+
+    try {
+      await sdk.txs.send({ txs: transactions.map((d) => d.raw) }).catch(console.error);
```
Comment (dasanra) [inline apps/tx-builder/src/components/Dashboard.tsx:95]: As convention we use full names even for this cases. I supose this is a transaction but as I didn't check in deep is not clear to me with only `d`

Don't worry to use longer names

**Review by dasanra — CHANGES_REQUESTED** [review]

File: apps/tx-builder/src/components/Dashboard.tsx (hunk 1 above)
Comment (yagopv) [reply apps/tx-builder/src/components/Dashboard.tsx]: I was moving previous code around and didn't see this. Done!!

**Review by dasanra — APPROVED** [review]
A really useful improvement! Thank you!

**Comment by JagoFigueroa** [comment]
All good, thank you Yago ;)

**Pull request merged**